test/MultitokenCollector.test.ts

```diff
@@ -0,0 +1,409 @@
+import {
+  FakeContract,
+  MockContract,
+  MockContractFactory,
+  smock,
+} from '@defi-wonderland/smock';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { ethers } from 'hardhat';
+import { Wallet, BigNumber } from 'ethers';
+import {
+  ERC20,
+  MultitokenCollector,
+  MultitokenCollector__factory,
+} from '../typechain-types';
+
+chai.use(chaiAsPromised);
+
+const expect = chai.expect;
+
+const PARTNER_SHARES = [20, 30, 40, 10];
+
+type Partner = {
+  beneficiary: string;
+  share: number;
+};
+
+const buildPartners = async (
+  partnerWallets: SignerWithAddress[],
+  shares: number[]
+) => {
+  return Promise.all(
+    partnerWallets.map(async (wallet, index) => {
+      const partner: Partner = {
+        beneficiary: await wallet.getAddress(),
+        share: shares[index],
+      };
+
+      return partner;
+    })
+  );
+};
+
+type CollectorDeployParams = {
+  ownerAddr?: string;
+  tokens?: string[];
+  partners?: Partner[];
+  remainderDestinationAddr?: string;
+};
+
+type CollectorDeployFunction = (
+  collectorParams: CollectorDeployParams
+) => ReturnType<MockContractFactory<MultitokenCollector__factory>['deploy']>;
+
+const prepareDefaultDeployer =
+  (
+    collectorFactory: MockContractFactory<MultitokenCollector__factory>,
+    defaultOwnerAddr: string,
+    defaultTokens: string[],
+    defaultPartners: Partner[],
+    defaultRemainderDestinationAddr: string
+  ): CollectorDeployFunction =>
+  ({
+    ownerAddr = defaultOwnerAddr,
+    tokens = defaultTokens,
+    partners = defaultPartners,
+    remainderDestinationAddr = defaultRemainderDestinationAddr,
+  }: CollectorDeployParams) =>
+    collectorFactory.deploy(
+      ownerAddr,
+      tokens,
+      partners,
+      remainderDestinationAddr
+    );
+
+describe('MultitokenCollector', function () {
+  let owner: SignerWithAddress;
+  let collectorFactory: MockContractFactory<MultitokenCollector__factory>;
+  let fakeERC20Tokens: FakeContract<ERC20>[];
+  let partners: Partner[];
+  let remainderDestination: SignerWithAddress;
+  let partner1: SignerWithAddress;
+  let partner2: SignerWithAddress;
+  let partner3: SignerWithAddress;
+  let partner4: SignerWithAddress;
+
+  beforeEach(async function () {
+    [owner, partner1, partner2, partner3, partner4, remainderDestination] =
```

Comment:
`Type 'SignerWithAddress | undefined' is not assignable to type 'SignerWithAddress'.
  Type 'undefined' is not assignable to type 'SignerWithAddress'`

Reply:
Added
```
    /*
     * Adding `undefined` to array item type makes less sense 
     * in testing, where items of arrays in runtime are 
     * generally guaranteed to exist, thus we overwrite
     * `noUncheckedIndexedAccess` to false.
     */
    "noUncheckedIndexedAccess": false,
```
to tsconfig